packages/runtime-common/module-syntax.ts

```diff
@@ -193,30 +196,76 @@ export class ModuleSyntax {
-    let cardClass: PossibleCardOrFieldClass | undefined;
-    if (card.type === 'exportedName') {
-      cardClass = this.possibleCardsOrFields.find(
-        (c) => c.exportedAs === cardName,
+  // This function performs the same job as
+  // @cardstack/runtime-common/code-ref.ts#loadCard() but using syntax instead
+  // of running code
+  private getCard(codeRef: CodeRef): PossibleCardOrFieldClass {
+    let cardOrFieldClass: PossibleCardOrFieldClass | undefined;
+    if (!('type' in codeRef)) {
+      cardOrFieldClass = this.possibleCardsOrFields.find(
+        (c) => c.exportedAs === codeRef.name,
       );
-    } else {
-      cardClass = this.possibleCardsOrFields.find(
-        (c) => c.localName === cardName,
+    } else if (codeRef.type === 'ancestorOf') {
+      let classRef = this.getCard(codeRef.card).super;
+      if (!classRef) {
+        throw new Error(
+          `Could not determine the ancestor of ${JSON.stringify(
+            codeRef,
+          )} in module ${this.url.href}`,
+        );
+      }
+      cardOrFieldClass = this.getPossibleCardForClassReference(classRef);
+    } else if (codeRef.type === 'fieldOf') {
+      let parentCard = this.getCard(codeRef.card);
+      let field = parentCard.possibleFields.get(codeRef.field);
+      if (!field) {
+        throw new Error(
+          `interior card ${JSON.stringify(codeRef)} has no field '${
+            codeRef.field
+          }' in module ${this.url.href}`,
+        );
+      }
+      cardOrFieldClass = this.getPossibleCardForClassReference(field.card);
+    }
+    if (!cardOrFieldClass) {
+      throw new Error(
+        `cannot find card ${JSON.stringify(codeRef)} in module ${
+          this.url.href
+        }`,
       );
     }
-    if (!cardClass) {
+    return cardOrFieldClass;
+  }
+
+  private getPossibleCardForClassReference(
+    classRef: ClassReference,
+  ): PossibleCardOrFieldClass | undefined {
+    if (classRef.type === 'external') {
+      if (
+        trimExecutableExtension(new URL(classRef.module, this.url)) === this.url
+      ) {
+        return this.possibleCardsOrFields.find(
+          (c) => c.exportedAs === classRef.name,
+        );
+      }
       throw new Error(
-        `cannot find card with ${startCase(
-          card.type,
-        ).toLowerCase()} of "${cardName}" in module`,
+        `Don't know how to resolve external class reference ${JSON.stringify(
+          classRef,
+        )} into a card/field. Module syntax only has knowledge of this particular module ${
+          this.url.href
+        }.`,
+      );
+    } else {
+      if (classRef.classIndex == null) {
+        throw new Error(
+          `Cannot resolve class reference with undefined 'classIndex' when looking up interior card/field in module ${this.url.href}`,
+        );
+      }
+      let declaration = this.declarations[classRef.classIndex];
+      return this.possibleCardsOrFields.find(
+        (c) => c.path === declaration.path,
```

Comment:
hmmmm. Is this code path being reached? 

In any case, classIndex is the index of the card of the possibleCardsOrFields array. Think we should just reference the possibleCardsOrFields array.

Reply:
It totally is—this code path is what is executed when the code ref is ancestorOf and fieldOf. You can see this in the module syntax unit tests, as well as that is literally what is being demoed in the screen capture

Reply:
I think you are getting the `declaration` confused with the `possibleCardOrField`. These are different things. If you follow the types, you’ll see that the syntax analysis refers to declarations for the super card or the field card, but what we actually want to return is the possibleCardOrField that corresponds to the declaration—that’s what getCard() is supposed to do.

Reply:
Or actually looking at this more closely I think you are right, I’m the one getting confused—I think im not trying this on a complex enough module. All the modules I’m trying this on only have declarations of classes. I’ll add another unit test that includes non class  declarations too that appear before the cards.